pkg/ring/lifecycler.go

```diff
@@ -564,11 +596,12 @@ func (i *Lifecycler) autoJoin(ctx context.Context, targetState IngesterState) er
 
 		newTokens := GenerateTokens(i.cfg.NumTokens-len(myTokens), takenTokens)
 		i.setState(targetState)
+
 		ringDesc.AddIngester(i.ID, i.Addr, newTokens, i.GetState(), i.cfg.NormaliseTokens)
```

Comment:
We only write newly generated `newTokens` to the ring, but if `myTokens` wasn't empty, there is a mismatch between what's in the ring, and what we keep in memory (`myTokens`).

(This is old bug, but since you're changing this code, perhaps we should fix it)

Reply:
Last time I checked, `AddIngester` appended the passed tokens to the existing tokens and didn't overwrite the old ones. But looking at the code I see it is appended in one case and replace in another https://github.com/cortexproject/cortex/blob/41d37d8f58c8ca5ba0ea44666fcac8e2a90e5290/pkg/ring/model.go#L38-L63

Reply:
You're right. This is a bug only when using normalization. I'll fix it in #1787 then.